### DISCUSSION TITLE
DataLoader cant load batch sizes bigger than 10

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4449] [Q&A] [answered]
If I set a batch_size bigger than 10, the dataloader will only load 10 graphs.

![image](https://user-images.githubusercontent.com/51344466/162626514-82f4e29a-a271-4208-98f4-7d5d0599e893.png)

**Comment by rusty1s** [comment, marked as answer]
This can only happen in case your dataset only contains 10 graphs.

**Reply by DogitoErgoSum** [reply]
i think i have more than that. Am i passing the folder path right?
![image](https://user-images.githubusercontent.com/51344466/162635810-44914b33-672c-4fa3-81b7-2f68f87221b7.png)

![image](https://user-images.githubusercontent.com/51344466/162635827-0a0e2af7-4021-4735-8d54-48ce64b067fd.png)

**Reply by rusty1s** [reply]
What does `len(dataset)` print out? Can you share more information about how you load in the dataset?

**Reply by DogitoErgoSum** [reply]
![image](https://user-images.githubusercontent.com/51344466/163412271-90b58e07-a093-49db-a783-82464fd9437f.png)

![image](https://user-images.githubusercontent.com/51344466/163412507-32ef20ca-e73e-444b-94fe-d5f876f704cf.png)


![image](https://user-images.githubusercontent.com/51344466/163412540-1d31bb7b-e7c3-4005-9bcb-0bbba6d1850e.png)

![image](https://user-images.githubusercontent.com/51344466/163412918-9654e20a-299d-4730-aec9-f0370bdca6f2.png)

**Reply by rusty1s** [reply]
Note that you set the length of the dataset to the length of the string „datallll_0.pt“ (=13) instead of the real number of examples. Just returning 30 inside `len()` should fix this issue.
